biocrnpyler/mechanism.py

```diff
@@ -331,11 +332,111 @@ def update_reactions(self, s1, s2, component = None, kb = None, ku = None, part_
             ku = component.get_parameter("ku", part_id = part_id, mechanism = self)
         if component == None and (kb == None or ku == None):
             raise ValueError("Must pass in a Component or values for kb, ku.")
-
-        complex = ComplexSpecies([s1, s2])
+        if(complex==None):
```

Comment:
Hi there,
I'd suggest that to test complex object points to None use this instead: "complex is None", because if someone implements the equality operator on the Complex class then this code will cause trouble. 
This is true for other cases where you test for None.
More info here:
https://stackoverflow.com/questions/3257919/what-is-the-difference-between-is-none-and-none